controllers/periodic_healthcheck.go

```diff
@@ -47,7 +47,7 @@ func (r *EtcdadmClusterReconciler) startHealthCheckLoop(ctx context.Context, don
 			return
 		case <-ticker.C:
 			etcdClusters := &etcdv1.EtcdadmClusterList{}
-			err := r.Client.List(ctx, etcdClusters)
+			err := r.uncachedClient.List(ctx, etcdClusters)
```

Comment:
what was the issue here with the cached client?

Reply:
the default client is using cached objects which might not be accurate especially with concurrent workers. This is mainly a precautionary change. I didn't see any apparent issues with using cached client either but it's could be because it's a hard case to run into. Also, changing it to uncached client didn't have any performance issues either.

Reply:
Got you. I would vote to move it back to cached then. This is a periodic check, so even if it gets an outdated list of clusters, the next periodic check will get a refreshed list.

Reply:
okay makes sense, I have changed it back to the regular client